linked_list.py

```diff
@@ -0,0 +1,348 @@
+
+# Defines a node in the singly linked list
+class Node(object):
+    def __init__(self, value=None, next_node=None):
+        self.__data = None  # protected
+        self.next_node = next_node
+        self.__set_data_once(value)
+
+    def get_data(self):
+        if self.__data:
+            return self.__data
+        else:
+            return "Undefined"
+
+    def __set_data_once(self, value):
+        if self.__data == None:
+            self.__data = value
+        else:
+            return ("You may not modify data.  Please add a node.")
+
+# Defines the singly linked list
+class LinkedList(object):
+    def __init__(self):
+        self.__head = None # keep the head private. Not accessible outside this class
+
+    def __retrieve_head(self):
+        return self.head
+
+    def __set_head(self, new_value):
+        self.head = new_value
+
+    # method to add a new node with the specific data value in the linked list
+    # insert the new node at the beginning of the linked list
+    def insert(self, data):
+        new_node = Node(value = data)
+        new_node.next_node = self.__head
+        self.__head = new_node
+        print ("inserting {}".format(data))
+
+    # method to find if the linked list contains a node with specified value
+    # returns true if found, false otherwise
+    def search(value):
+        pass
+
+    
+
+    # method to return the max value in the linked list
+    # returns the data value and not the node
+    def find_max(self):
+        if self.__head == None:
+            return "None"
+
+        node = self.__head
+        max = node.get_data()
+        while node.next_node != None:

```

Comment:
Should this be `node != None:` instead of `node.next_node != None:` ? Check if you get the right result when the last node in the linked list has the max value.